docs/guides/vscode.md

```diff
@@ -155,11 +155,32 @@ The VSCode extension is based on a [language server](https://en.wikipedia.org/wi
 
 If you have environment variables that are needed by the context and the language server, you can use one of these approaches to pass variables to the language server:
 
-- Open VSCode from a terminal that has the variables set
-- Use environment variables pulled from somewhere else dynamically (e.g. a `.env` file) in your config 
+- Open VSCode from a terminal that sets environment variables in its configuration file (e.g., `.bashrc` for Bash or `.zshrc` for Zsh). The variables will automatically be picked up by the language server only if you explicitly export them in the shell configuration file like `export ENV_VAR=value`.
+- Use environment variables pulled from somewhere else dynamically (e.g. a `.env` file) in your root project directory. This will automatically be picked up by the language server.
 - Set the environment variables in the python environment that the extension uses. You can find detailed instructions [here](https://code.visualstudio.com/docs/python/environments#_environment-variables)
```

Comment:
This isn't quite what was intended beforehand, but I'm happy to clarify it. 

1. Open VSCode from a terminal that has the variables set is not about opening shells that have the variables set, if you open VSCode from a terminal with env vars set, it inherits them. 

That's different to

2. The LSP server runs in your shell, and anything you have configured in there gets picked up 

3. You can manually pull in vars in a Python config file as you wish. 

4. `.env` gets pulled because of the Python environment setup. It's not a separate thing. That's how it's being pulled in

Reply:
Thanks! Updated for accuracy.